test/Interpreter/objc_extensions.swift

```diff
@@ -4,16 +4,54 @@
 
 import Foundation
 
+// Category on a nested class
 class OuterClass {
   class InnerClass: NSObject {}
 }
 
 extension OuterClass.InnerClass {
   @objc static let propertyInExtension = "foo"
+
+  @objc func dynamicMethod() -> String {
+    return "bar"
+  }
 }
 
 let x = OuterClass.InnerClass()
 
 // CHECK: foo
 print(type(of: x).propertyInExtension)
 
+// CHECK: bar
+print(x.dynamicMethod())
+
+// Category on a concrete subclass of a generic base class
+class Base<T> {
+  let t: T
+
+  init(t: T) { self.t = t }
+}
+
+class Derived : Base<Int> {}
+
+extension Derived {
+  @objc func dynamicMethod() -> Int {
+    return t
+  }
+}
+
+let y = Derived(t: 100)
+
+// CHECK: 100
+print(y.dynamicMethod())
```

Comment:
I'm not sure I trust this test to invoke the method dynamically. Can you add a respondsToSelector check?

```
// CHECK: respondsToSelector? true
print("respondsToSelector? \(y.responds(to: #selector(Derived.dynamicMethod)))")
```

Reply:
Well, `Derived` does not descend from `NSObject`, which was intentional (remember when we had the resilient class hack in 4.2, and I only did it for @objc classes, and it broke categories).

However, using AnyObject dispatch also clearly demonstrates that its dynamically dispatched.

And FWIW, we infer `dynamic` on `@objc` members of extensions, because you're supposed to be able to override them in subclasses.

Reply:
Yeah, I know it is actually testing all that, but I'd prefer it be a little more direct. AnyObject dispatch sounds good.